src/server.cpp

```diff
@@ -20,19 +23,55 @@ void Server::start_receive()
         std::bind(&Server::handle_receive, this, pls::_1, pls::_2));
 }
 
-void Server::handle_receive(const error_code& error, std::size_t /*bytes_transferred*/)
+void Server::handle_receive(const error_code& error, std::size_t bytes)
 {
-    if (!error || error == boost::asio::error::message_size)
+    if (!error)
     {
-        const std::string message("Hello, World!");
+        size_t length;
+        try
+        {
+            length = recv_buffer_.at(2) * 256 + recv_buffer_.at(3);
```

Comment:
`recv_buffer_` is 4096 bytes long. `at` function will never throw, no reason to use it instead of `operator[]`.

The question here (which I asked in previous review) is what if we receive only 1 byte of data? Values at `recv_buffer_[2]` and `recv_buffer_[3]` would be undefined, you would get some random number in `length`.

Before accessing any received data you should check that you actually received it. In this particular case you need to check that you received at least 4 bytes of data, that 0-3 bytes of buffer are actually filled with the data.

In fact, you can easily check that you received at least 20 bytes of data - the header and the authenticator. Anything less than that would be a malformed request.